### PR TITLE
Upgrade cosmos sdk to v0.46.0-rc1

### PR DESCRIPTION
**Pull request by DavidNix**
This also upgrades ibc-go to point to HEAD of [this branch](https://github.com/cosmos/ibc-go/tree/carlos/upgrade-sdk-0.46-tendermint-0.35).

This resolved an issue earlier where ibc-go was relying on a package removed from the sdk rc.

### CODE REVIEW

File: router/keeper/keeper.go [hunk 1]
```diff
@@ -61,7 +61,7 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 
 func (k Keeper) ForwardTransferPacket(ctx sdk.Context, parsedReceiver *parser.ParsedReceiver, token sdk.Coin, labels []metrics.Label) error {
 	var err error
-	feeAmount := token.Amount.ToDec().Mul(k.GetFeePercentage(ctx)).RoundInt()
```
Comment (DavidNix) [inline router/keeper/keeper.go:64]: `ToDec()` was removed.

File: router/module.go [hunk 2]
```diff
@@ -261,7 +262,7 @@ func (am AppModule) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, re
 			prefixedDenom := transfertypes.GetDenomPrefix(packet.GetDestPort(), packet.GetDestChannel()) + newData.Denom
 			denom = transfertypes.ParseDenomTrace(prefixedDenom).IBCDenom()
 		}
-		unit, err := sdk.ParseUint(newData.Amount)
+		unit, err := math.ParseUint(newData.Amount)
```
Comment (DavidNix) [inline router/module.go:265]: The rc moved this function into a separate `math` package which, as it appears, is released as a separate go module - `cosmossdk.io/math`.

**Comment by DavidNix** [comment]
@jtieri Ready for review 👀 . 

If looks ok, I'll merge and cut another release. Hopefully gaia team can use this one.

@okwme fyi ☝️

**Review by jtieri — APPROVED** [review]
🚀

**Pull request merged**